slot2 analog wip

nadiaholmquist opened the pull request.

nadiaholmquist committed:
GBACart: implement Slot-2 analog stick device for SM64DS hack

nadiaholmquist committed:
do the deadzone per axis instead of using the magnitude like SM64

Backflips are almost impossible to pull off in SM64DS if you don't do this

redbluegreenhat committed:
Set SDL_HINT_APP_NAME (#2319)

Fixes #2300

redbluegreenhat committed:
Use standard sysconf

Fixes compilation of JIT builds on non-glibc OSes. After some testing in a Fedora 41 VM,
__sysconf and sysconf return the same value, and sysconf in glibc appears to just
be an alias to __sysconf to begin with

JesseTG committed:
Fix undefined behavior when indexing into `ARCode::Code` (#2331)

- Indexing past the end of a `std::vector`'s length is undefined, even if there's extra capacity
- GCC 15 introduced an assert in `vector::operator[]`, so this line caused an abort if melonDS was built with GCC 15
- It was always undefined, but now the STL checks for it

JesseTG committed:
Fix a missing `#include` on Windows builds (#2333)

* Fix a GCC 15 build issue on Windows due to a missing `#include`

- `<vector>` was included implicitly by some other header
- The build broke in GCC 15 on MinGW, most likely due to some internal refactoring

* Indent these `#include`s the same as the others

Arisotura committed:
add source for DLDI driver

Arisotura committed:
update copyright headers (about time)

Arisotura committed:
melonDLDI: add support for unaligned I/O

Arisotura committed:
fix mic settings not being changed when closing the audio settings dialog

nadiaholmquist committed:
work around building with LTO causing an ICE in gcc 15.1.0

Arisotura committed:
multi-instance: load firmware from correct instance-specific file (load from original file if not found)

Jaklyy committed:
small fix to translucency flag assignment (#2301)

hardware does not care that the polygon's mode would prevent the texture from rendering translucent pixels

nadiaholmquist committed:
Don't try to change the core's audio interp setting when the emu
instance has no core yet.

Fixes #2352

Arisotura committed:
setupMicInputData() ought to be called at all times tho

edo9300 committed:
Load Tad key into DSi AES engine (#2252)

asiekierka committed:
Slot-2 Motion Pak, Guitar Grip emulation (#2183)

* Add DS Motion Pak emulation

* Add retail Motion Pak emulation, Guitar Grip emulation

* Simplify Motion Pak acceleration conversion formula

* Fix Motion Pak emulation axes

* Motion Pak: Emulate console laying on a flat table when motion input is not detected

* Motion Pak: Add comment

* GBACartMotionPak: Update comment

Arisotura committed:
DSP: fix bug in PDATA read DMA (was reading wrong register)

Arisotura committed:
camera: trigger DMA when reaching the end of a frame (fixes issues when the frame height isn't a multiple of the DMA interval)

edo9300 committed:
Don't return 0 when attempting to read from the GPA GPIO addresses with GPIO disabled (#2266)

Arisotura committed:
DSP: fix PDATA reads being one off

Arisotura committed:
Merge remote-tracking branch 'origin/master'

Arisotura committed:
huh

Arisotura committed:
NDMA: fix IRQ in infinite repeat mode

Arisotura committed:
camera: fix resolution selection code (could accidentally select a resolution like 640x360)

Arisotura committed:
fix issues with multi-window and OpenGL on Windows

Arisotura committed:
window: move cleanup code to destructor (closeEvent() isn't called for children)

Arisotura committed:
window: undo last commit (was a trainwreck). explicitly close child windows to avoid GL issues. fix bug with window parenting in second multiplayer instances.

Arisotura committed:
fix video settings update on second instance secondary windows

Arisotura committed:
add a way to tell windows apart

Arisotura committed:
call MakeCurrent() before deiniting GL stuff (to avoid deiniting on the wrong context)

Arisotura committed:
camera: make timings more realistic (fixes rolling in some games)

Arisotura committed:
camera: model FIFO more accurately (avoids weird bugs when DMA fails)

Arisotura committed:
DSi: add support for SCFG_EXT bit 13 (32-bit VRAM bus)

Arisotura committed:
Revert "camera: model FIFO more accurately (avoids weird bugs when DMA fails)"

This reverts commit 2cb07bf126fba0d6d8960a1b2e23486809c067d0.

Arisotura committed:
push this back tho...

Arisotura committed:
camera: revise interface behavior to be more accurate
* there are two FIFO buffers (finally fixes Let's Golf)
* fix issues with camera start condition/cnt bit15
* add camera interface state to savestate

Arisotura committed:
camera: fix some misc issues

Arisotura committed:
oops

Arisotura committed:
camera: misc fix

Sparronator9999 committed:
Make SPU audio single-buffered (audio latency improvement) (#2286)

* SPU audio latency improvements

Basically reverts audio buffer handling to what it was before commit 05b94ef, but with the mutexes kept for thread safety (which the referenced commit was trying to do).

The SPU audio buffer should still be thread-safe in theory... right?

* Audio output improvements

This commit changes the audio output buffer to be configured by a variable, and fixes the case where the sound driver may change the buffer size after calling SDL_OpenAudioDevice (e.g. if the buffer size is set too low for the driver to handle).

Jaklyy committed:
fix full bios boot not being set properly on initial DSI class creation (#2366)

Arisotura committed:
BAHAHAHAAHAHAHAHAAHAHAHA-*~°+|

KostaSaizo7 committed:
le plus important (#2375)

blarg

Izder456 committed:
sockaddr_in is a forward declaration on *BSD, explicitly include the correct header

Arisotura committed:
(finally) change software renderer to use 32-bit division

FrankHB committed:
Fixed 8-bit NO$GBA debug register write. (#2399)

Signed-off-by: FrankHB <frankhb1989@gmail.com>

CasualPokePlayer committed:
Fix FF_ReadStorage/FF_WriteStorage callback return value (#2400)

The return value for these callbacks is in sectors, not bytes.

CasualPokePlayer committed:
Fix save type for Legacy of Ys - Books I & II (#2391)

Reported on Discord for BizHawk that going into Ys II after doing Ys I would result in a corrupted save. DeSmuMe reportedly doesn't have this issue, and it has a 64KiB save instead.

GABO1423 committed:
Windows: Quick and Dirty Build Fix (#2401)

An upper version limit should be set here again when there is a new GCC release that fixes the bug.

RayyanAnsari committed:
ROMInfoDialog: make DS banner more accurate

Change some styling to make the DS banner box in the ROM Info
dialog look more like the real banner box in the BIOS by:

- Adding the stripes inside the text box
- Adding the banded background in the box as a whole
- Adding the inner white border
- Disabling anti-aliasing on the title font
- Reducing letter spacing and line height

Arisotura committed:
DSP HLE (#2404)

* DSP HLE (support for AAC, G711 and Graphics ucodes)
* DSP audio output and mic input (both HLE and LLE)
* DSi I2S interface and mic input
* smarter mic input system

Arisotura committed:
add faad dep to build instructions

Arisotura committed:
blarg

Arisotura committed:
make input system thread-safe

Arisotura committed:
doesn't seem to be a good change

Arisotura committed:
crap

nadiaholmquist committed:
nix: add faad2 dependency, update packages while we're at it

nadiaholmquist committed:
ci: vcpkg improvements

* Upgrade to 2025.07.25
* Configure vcpkg settings in vcpkg.json, this avoids needing to
  duplicate the vcpkg hash used in several places, and also avoids
  needing to set overlay ports/triplets directories from CMake
* Add caching workaround so that builds won't take forever every time
* Build on windows-2025

RayyanAnsari committed:
JIT: x64: ensure writable code memory after alloc

Call JitEnableWrite() immediately following the allocation of the JIT
code memory in the x64 compiler to make certain that it is in RW mode.

This call is made following allocation in the A64 compiler already, see
9ad3d42 ("hopefully fix macos").

Switching the memory protection between RW and RX is required on systems
that enforce an executable-space protection policy, such as NetBSD and
OpenBSD, with PaX MPROTECT and W^X respectively.

However, adherence to these policies for these platforms is not
implemented yet, so at this moment in time this call should be a no-op
on x64.

Arisotura committed:
Revert "Fix FF_ReadStorage/FF_WriteStorage callback return value (#2400)" (#2421)

This reverts commit 18bffde207380039256b7a53306b6a448cfcac83.

RayyanAnsari committed:
JIT: fastmem: handle W^X codemem in fault handler

Set the protection on the code memory to RW when the fault handler
rewrites code, and set it back to RX, the protection prior to the
interrupt, after doing so.

This fixes a segfault that occurred on NetBSD, due to its PaX MPROTECT
policy, when trying to run fastmem.

This could potentially be the reason as to why fastmem does not work on
Apple Silicon Macs, although I am unable to confirm as I do not possess
such a device to test with.

RayyanAnsari committed:
JIT: add support for PaX MPROTECT on NetBSD

Call mprotect to switch the code memory protection between RW and RX
when needed on NetBSD, which is required due to PaX MPROTECT being
enabled on all binaries by default by the operating system.

This removes the need to run `paxctl +m melonDS`.

This patch was tested on a x64 VM running NetBSD 10.1. I can confirm
that this fixes JIT on NetBSD, however fastmem does not work as of yet
due to a segfault.

Although invoking mprotect does incur a performance penalty, it is still
faster than running without JIT.

A potentially faster alternative is to back the JIT code memory with an
anonymous file descriptor in physical memory, then map this physical
memory to two virtual memory addresses: one with RW permissions for code
generation, the other with RX permissions for code execution.

Instead of using platform-specific methods to create the anonymous file
descriptor, such as memfd_create on Linux, we could copy a 'trick' that
we use in fastmem already: calling shm_open, a POSIX standard, with a
random name, and immediately unlinking it.

However, this would add some more complexity and would be a bit harder
to implement. For now, this works fine. Perhaps another time...

RayyanAnsari committed:
NDS: call JIT.Reset() instead of JIT.ResetBlockCache()

Call the reset function of the JIT object instead of resetting the block
cache directly when disabling the JIT compiler, to ensure that the
protection on the code memory is switched to RW as required.

It is imperative that calls to ResetBlockCache() directly must be
preceeded by JitEnableWrite() to comply with executable-space protection
on systems that require it.

This fixes a segfault on NetBSD, due its PaX MPROTECT policy, that
occured when disabling JIT during runtime.

Izder456 committed:
JIT: fix building with ENABLE_JIT on OpenBSD targets (#2398)

* sockaddr_in is a forward declaration on *BSD, explicitly include the correct header

* fix building with ENABLE_JIT on OpenBSD targets

* add wxallowed linker flag for openbsd targets

* OpenBSD armjit sc_exr -> sc_elr

* wxallowed -> wxneeded

* force mmap for jit on OpenBSD

* disable fastmem on OpenBSD

* fix typo in disabling fastmem for OpenBSD

* add missing space in this if

Arisotura committed:
fix possible crash in SD/MMC code

jonko0493 committed:
fix: GDB stub reading wrong code for vCont packet (#2388)

* fix: GDB stub reading wrong code for vCont packet

* Implementation did contain bug, fixup

Izder456 committed:
Create OpenBSD CI using vm-actions (#2423)

Arisotura committed:
update log2(size) in melonDLDI (it's more than 512 bytes long)

RayyanAnsari committed:
ci: add x86-64 FreeBSD and NetBSD jobs

Adapt the existing OpenBSD workflow to also run on FreeBSD and NetBSD,
using the dynamic-uses action and a matrix strategy for each OS.

Arisotura committed:
merp

Arisotura committed:
fix DSi camera reset behavior (CAM_MCNT)

fixes #2437

Arisotura committed:
huh. oops

RayyanAnsari committed:
README: add badge for (Free|Net|Open)BSD workflow

RayyanAnsari committed:
FIFO: don't drop const qualifier in cast

Cast the data pointer to const u8*, not u8*, to fix a compiler warning.

nadiaholmquist committed:
Add high-quality resampling using blip-buf to the core (#2412)

* Add high-quality resampling using blip-buf to the core

This also gets rid of the need for the frontend to resample the audio,
as it can simply ask the core to produce audio at whatever sample rate
it desires.

* allow setting an audio output skew to alter the timing slightly
  This makes up for the difference of e.g. the frontend targeting a clean
  60FPS instead of the actual 59.8...FPS.

* remove hack that was used to have the SPU produce silence

Arisotura committed:
fix UB in software renderer

Arisotura committed:
AR: add fun D4 opcodes

nadiaholmquist committed:
qt6: use new way to access the native display/window on X11/Wayland

This removes the need to include private headers from QtGui, which
should remove the need to install the relevant package and make melonDS
easier to build in nonstandard environments.

nadiaholmquist committed:
nix: update inputs and add dev shell Qt wrapper workaround

nadiaholmquist committed:
fix the qt version check so it actually works

Arisotura committed:
DSi: add hack so we can properly adjust main RAM size when loading a DS ROM

Arisotura committed:
Merge remote-tracking branch 'origin/master'

Arisotura committed:
remove unnecessary printfs

nadiaholmquist committed:
nix: update inputs

fixes the Qt dependency on macOS

Arisotura committed:
AR cheat database import feature (#2479)

* support for importing cheats from usrcheat.dat
* expand cheat format to add support for the extra info in usrcheat.dat
* add said features to the cheat editor
* improve cheat editor UI (add support for moving stuff around, etc)

Arisotura committed:
fix Wayland OpenGL glitchiness

Jaklyy committed:
fix minor thumb blx immediate bugs (#2450)

JordanTheToaster committed:
SRC: Update xxHash to 0.8.3 (#2406)

nadiaholmquist committed:
reduce the hardcoded frontend audio buffer size to 512

seems to fix crackling on Windows

nadiaholmquist committed:
unbreak audio sync

nadiaholmquist committed:
initialize curFPS to a reasonable value

Fixes audioGetNumSamplesOut sometimes breaking audio completely on start
because of using the value uninitialized.

Full credit to Arisotura for somehow figuring this one out

FireNX70 committed:
Implement tile size scaling for the compute renderer (#2065)

* Fix glMemoryBarrier flags

* Scale TileSize with internal resolution

* Clean up tile size calc

* Use a different CoarseTileCountY with size 32 tiles

* Better tile scaling calc

* Actually good way to get the MS bit

* Tie BinCombined's local_size_x to the coarse tiles' area

* Work count X is unrelated to the local size here

* Adjust ClearCoarseBinMask's local size according to TileSize

* Add extra barrier for radeonsi.

nadiaholmquist committed:
SPU: make sure output buffer is properly initialized to a null pointer

fixes potential crash on boot

nadiaholmquist committed:
Windows non-MinGW Clang support (#2429)

* Basic support for building with Visual Studio/clang-cl

Clang is required, MSVC support needs too many code changes.
ARM64 doesn't seem to work at the moment because of a bug in Qt.

* Fix close being used instead of closesocket in GdbStub in Windows

The GdbStub was closing sockets on windows by using "close" rather than "closesocket"

* fix the gdbstub unconditionally including unistd.h on windows and make it default-on again

* detect VS toolchain and arm64 with vcpkg, also get rid of some warnings
on Windows

* fix Qt indexOf ambiguity

* stupid math library issues

* add _USE_MATH_DEFINES so M_PI exists
* work around faad2 uncondtionally linking libm even on Windows where it doesn't exist

* Fix JIT linkage code for x86-64 in clang-cl builds

* Make it build with non-cl clang as well

* fix closesocket change breaking on macOS due to missing unistd include

* little build cleanups

---------

Co-authored-by: Edoardo Lolletti <edoardo762@gmail.com>
Co-authored-by: Gloria Goertz <gloria@gloriaaaa.eu>

RayyanAnsari committed:
core: Fix xxhash compile errors when using -Og

RayyanAnsari committed:
frontend: CheatImport: fix duplicate widget name warning

nadiaholmquist committed:
temporarily disable JIT on x86_64 macOS

Arisotura committed:
BAHAHAAAHAHAHAHARKGKGKJDSJD----%~^`!*

Arisotura committed:
greater delay for stopping NDS mic (PSG_Play.nds being weird)

Arisotura committed:
attempt at fixing this shit

Arisotura committed:
TSC: properly clamp output to 12 bits

nadiaholmquist committed:
Assorted Windows non-MinGW QoL improvements

* Add presets so Visual Studio will have reasonable build configurations
  by  default
* Fetch vcpkg with git manually so CMake won't try to overwrite it and
  delete all your already-built packages
* Support cross-compiling for Windows arm64 from x86_64

nadiaholmquist committed:
ARMJIT_Linkage.S: Just use the correct _WIN64 instead of that workaround

It works for both the MinGW and MSVC toolchains.

nadiaholmquist committed:
cmake never fails to surprise

RSDuck committed:
fix #2493

RSDuck committed:
replace remaining uses of fopen with Platform::OpenFile
should prevent code page issues on windows

Rokim21 commented:
Ive tried to use this version of MelonDS, alongside the Solatorobo patch in shoco, but it freezes the game when I try it even with slot-2 loaded in.
Any ideas why that may happen?

nadiaholmquist commented:
What patch is that exactly? The only thing I have tried this with is the patched English version of Super Mario 64 DS, I have no idea if it works with anything else.

Arisotura committed:
NDS: enable ITCM during direct boot (fixes armwrestler)

Rokim21 commented:
> What patch is that exactly? The only thing I have tried this with is the patched English version of Super Mario 64 DS, I have no idea if it works with anything else.

It's within the same thread as the one with the original SM64DS patch, same author and all

nadiaholmquist committed:
buffer audio at a regular interval instead of at the end of frames

This seems to help a lot with the stability of audio playback in things
like the stock DSi apps, I suspect uneven frame times might have been
throwing off the way it was done before.

nadiaholmquist committed:
flush any remaining audio samples to the output buffer on frame end

ensures that all audio for the frame is actually available for instance
in cases where audio is not running on a separate thread

RSDuck committed:
fix #2497

Paegasus committed:
DateTimeDialog: update txtNewCustomTime to current time when enabling 'Change to:' checkbox (#2490)

plenfulcleadent committed:
fix building with disable wayland (#2513)

QWaylandApplication definition doesn't exist on wayland disabled qt
binaries

Ndymario committed:
Fix macOS linking* (#2527)

asiekierka committed:
Use intrinsics to optimize bit counting operations (#2531)

Co-authored-by: Jaklyy <102590697+Jaklyy@users.noreply.github.com>

asiekierka committed:
SD/MMC controller fixes (#2438)

* Fix single block writes not being handled correctly if data was not already in FIFO

* Set the default block size for the sdmmc controller to 512

This is the assumption that profi's sdmmc driver makes, which follows the SD specs and properly behave on hardware, with this, his nand driver (and by extension the one used by blocksds) properly works

* Clean up MMC constants

---------

Co-authored-by: Edoardo Lolletti <edoardo762@gmail.com>

RayyanAnsari committed:
ROMInfoDialog: fix crash when ROM has no banner

It seems that at some point, the banner fetch function was changed to
return a nullptr instead of a zero-initialised banner for ROMs that did
not have one.
This caused a crash when opening the ROM info dialog due to a null
pointer dereference for said ROMs.
Fix this by handling the case where the banner is null.

Whilst we're here, also split out the header and banner info population
into their own methods to make the constructor a bit nicer to read.

Fueziwa committed:
fix Swap Screen Emphasis" hotkey bug (#2544)

nico-abram committed:
Audio mute options (#2504)

* Audio mute toggle hotkey

* Mute while fastforwarding option

* Fix capitalization in "Toggle audio mute"

guidocella committed:
make the swap emphasis hotkey update the checked screen sizing (#2540)

danielfikko committed:
ci: add wayland support to AppImage releases (#2547)

Arisotura committed:
start with SPU muted (POWCNT2=0)
fixes loud pops when booting firmware

Arisotura committed:
blackmagic3 - brand new OpenGL 2D renderer (#2553)

* reworked renderer structure to more closely reflect DS hardware, and have a general "renderer" object instead of two separate ones
* new OpenGL 2D renderer - hi-res display capture, hi-res rotscale
* 3D renderers support hi-res display capture as well
* 3D: clear bitmap support
* 2D: extra accuracy (extra green bit, mosaic fixes, forced blank fixes)
* more accurate video timing state machine
* support for POWCNT1 bit 0 (LCD enable)

closes #421, closes #473, closes #568, closes #1851, closes #1242, closes #1510, closes #1544, closes #2348

PoroCYon committed:
fix "-b always" not having any effect (#2565)

Izder456 committed:
Replace _POSIX_SOURCE with _POSIX_C_SOURCE 200809L in ffsystem.c (#2562)

This requests a modern POSIX interface set which reflects modern platform
realities. This ensures localtime_r(3) is declared on modern POSIX compliant
platforms such as the BSDs, as well as silences a "localtime_r not declared"
warning.

Tested on OpenBSD/amd64 7.8-stable.

PoroCYon committed:
fix BIOSPROT initialization (#2564)

Arisotura committed:
add basic sanity checks to ROM loading

Arisotura committed:
automate "full BIOS boot" based on BIOS checksums

Arisotura committed:
remove "separate savefiles" option. weird and unintuitive.

Arisotura committed:
blarg

asiekierka committed:
Allow booting in DSi direct mode without BIOS/firmware/NAND dumps. (#2523)

* Allow booting in DSi direct mode without BIOS/firmware/NAND dumps.

This is an initial patch which enables booting programs in DSi direct mode
without requiring BIOS/firmware/NAND dumps from increasingly expensive
consoles:

- The open DSi BIOS is derived from a modified build of FreeBIOS; minor patches
  have been done (such as moving SWI 0x12 to 0x02), but no new SWI functions
  have been implemented.
- The DSi firmware stub is generated in the same way as the NDS firmware stub;
  this appears to be good enough for BlocksDS's dswifi to pick up the new AP
  definitions.
- The DSi NAND is currently not stubbed at all; rather, the code has been
  edited to allow bringup without a NAND image. This is sufficient for DSi
  mode homebrew; programs running from the cartridge slot are likewise not
  expected to require NAND access.

More work has to be done before it's useful for non-homebrew scenarios.

* Implement a less stub DSi::NeedsDirectBoot()

* Do not block direct boots if DSi NAND is missing but BIOS/firmware is present

* Correctly initialize extended header for DSi generated firmware

* Fix overriding firmware language in DSi mode, add experimental support for Chinese language option

Arisotura committed:
fix up starting tab for EmuSettingsDialog (again)
thinking maybe we should set it programmatically so it stops being a problem

Arisotura committed:
update copyright headers

Arisotura committed:
ugh did I really accidentally commit this

nadiaholmquist committed:
nix: update flake inputs and increment version number

PoroCYon committed:
NWifi: add compatibility fixes for blocksds dswifi (#2596)

Arisotura committed:
Cartridge emulation improvements (#2595)

* refine cart interface implementation to match hardware, including timing fixes
* add second cart slot for DSi
* fully implement SCFG_MC
* move cart classes to subdirectory

fixes #2586
fixes #380 (might need extensive testing)
fixes Surviving High School (DSiWare)

Arisotura committed:
popular vote says this notation ain't good

Arisotura committed:
make this consistent

Arisotura committed:
fuck, CLion, is it too much to ask of you to respect existing indentation

Arisotura committed:
implement all SCFG_EXT access bits (and do it more cleanly)

Arisotura committed:
fix up SCFG_EXT values for direct boot to match what the DSi menu uses

Arisotura committed:
fix possible freeze during cart transfer (Dragon Ball 2)

Arisotura committed:
blarg

Arisotura committed:
fix GL renderer when using multiple instances

Jaklyy committed:
fix cart address wrappingnot being handled properly (#2606)

Mrlinkwii committed:
CI: update actions/{checkout,upload-artifact} to latest version (#2612)

* CI : update actions/checkout to latest version

* CI: update actions/upload-artifact from v4 to v7

---------

Co-authored-by: Rayyan Ansari <rayyan@ansari.sh>

RayyanAnsari committed:
CI: Fix FreeBSD hang (#2624)

Although it did work at some point, I think that FreeBSD's locking
over NFS is a bit unreliable, so we end up with Qt's automoc/autorcc
just hanging indefinitely. (some sort of deadlock?)

Fix this by building in /tmp inside the VM, and then copying it back
to the NFS-synced workspace when we are done.

Arisotura committed:
fix FreeBIOS IRQ wait (fixes #2600)

pretty sure a better fix for this is needed, the old code was broken, but atleast now it's broken the same way

Arisotura committed:
add RTC sync option (fixes #2589)

also allow to change date/time while game is running

nadiaholmquist committed:
vcpkg 2026.03.18

nadiaholmquist committed:
ci: Use the Windows SDK and Clang for Windows builds

Also, explicitly set the resource compiler as CMake seems to not find it
in some cases, and use the regular Clang/Clang++ for simplicity.

nadiaholmquist committed:
ci: Add arm64 builds for Windows

nadiaholmquist committed:
Attach to parent console process on Windows

This is directly based on the code for this from NanoBoyAdvance.

nadiaholmquist committed:
remove margin for filling in audio samples when running below full speed

this seems to reduce the amount of clicking/popping in the audio
somewhat

nadiaholmquist committed:
relax the limitation of the audio skew allowed for the target FPS

nadiaholmquist committed:
Fix occasional segfault on exit caused by emuIsActive being called after
the EmuThread is destroyed

Commit:
Fix SPU output buffer leak

nadiaholmquist committed:
Use Qt's QFile abstraction for the Platform.h file I/O functions (#2646)

* Use Qt's QFile abstraction for the Platform.h file I/O functions

* forgot to reimplement FileWriteFormatted

nadiaholmquist committed:
ci: avoid unnecessary extra archiving

actions/upload-artifact@v7 now supports disabling zipping the uploaded
artifact, so let's use that to avoid double-zipping the macOS app
bundle, as well as the AppImage as those should already be compressed.

nadiaholmquist committed:
Fix a couple of warnings reported by Clang

RSDuck committed:
fix #2657

nadiaholmquist committed:
ci: switch windows to windows-2025-vs2026

GitHub will be doing this automatically starting June 8, 2026, so just
get it over with now to ensure everything builds fine.

nadiaholmquist committed:
take care of various deprecateion and format warnings on windows

nadiaholmquist committed:
ROMInfoDialog: fix game title being unreadable with dark themes

nadiaholmquist committed:
Cleaner handling of termination signals on UNIX

nadiaholmquist committed:
ci: set fail-fast: false for macOS so one arch doesn't fail the other

nadiaholmquist committed:
nix: update inputs

nadiaholmquist committed:
vcpkg 9b965a116838c6cdcd36bca60d1b81b030c8ab8d

using latest commit from master as build fix for Qt 6 on macOS 26.4 was
not included in the latest tagged release

Arisotura committed:
fix possible bug with VRAMSTAT (good catch Jakly!)

dannyow committed:
wifi: don't auto power down the transceiver when power saving is disabled (#2681)

sleep mode 2 ("power saving disabled", used by dswifi in infrastructure
mode) was treated like automatic mode: when the one-shot W_BeaconCount2
countdown armed at wifi init (0xFFFF TUs = ~67s) reached zero, IRQ13
powered the transceiver off. stations running with power saving disabled
never service IRQ13/IRQ15, so nothing ever turned it back on, silently
killing all wifi traffic ~67 seconds after init. real hardware does not
power down in this mode.
